--- /Users/yumeng/PycharmProjects/SysEdit/23/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/23/r/after.java
@@ -1,1 +1,1 @@
-public RecoveredElement add(FieldDeclaration newFieldDeclaration, int bracketBalance) {
+public RecoveredElement add(FieldDeclaration newFieldDeclaration, int bracketBalanceValue) {
@@ -13,1 +13,1 @@
-            return this.parent.add(newFieldDeclaration, bracketBalance);
+            return this.parent.add(newFieldDeclaration, bracketBalanceValue);
@@ -25,1 +25,1 @@
-            return this.parent.add(newFieldDeclaration, bracketBalance);
+            return this.parent.add(newFieldDeclaration, bracketBalanceValue);
