--- /Users/yumeng/PycharmProjects/SysEdit/37/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/37/r/after.java
@@ -2,3 +2,3 @@
-    org.eclipse.compare.contentmergeviewer.IMergeViewerContentProvider cp = getMergeContentProvider();
-    if (cp instanceof org.eclipse.compare.internal.MergeViewerContentProvider) {
-        org.eclipse.compare.internal.MergeViewerContentProvider mcp = ((org.eclipse.compare.internal.MergeViewerContentProvider) (cp));
+    IMergeViewerContentProvider cp = getMergeContentProvider();
+    if (cp instanceof MergeViewerContentProvider) {
+        MergeViewerContentProvider mcp = ((MergeViewerContentProvider) (cp));
@@ -6,1 +6,1 @@
-            case 'A' :
+            case ANCESTOR_CONTRIBUTOR :
@@ -9,1 +9,1 @@
-            case 'L' :
+            case LEFT_CONTRIBUTOR :
@@ -12,1 +12,1 @@
-            case 'R' :
+            case RIGHT_CONTRIBUTOR :
