--- /Users/yumeng/PycharmProjects/SysEdit/16/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/16/l/after.java
@@ -14,1 +14,1 @@
-        throw new bsh.EvalError("reflection error: " + e, this);
+        throw new bsh.EvalError("reflection error: " + e, this, callstack);
@@ -16,1 +16,1 @@
-        throw new bsh.TargetError(e.getTargetException(), this);
+        throw new bsh.TargetError("target exception", e.getTargetException(), this, callstack, true);
