--- D:\expdata\c3\SysEdit\34/l/before.java
+++ D:\expdata\c3\SysEdit\34/l/after.java
@@ -124,0 +124,8 @@
+    IPreferenceStore store = getCompareConfiguration().getPreferenceStore();
+    if (store != null) {
+      if (store.getBoolean(ComparePreferencePage.SHOW_MORE_INFO)) {
+        statusLabel = new Label(parent, SWT.NONE);
+        statusLabel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+      }
+    }
+
