--- D:\expdata\c3\SysEdit\31/l/before.java
+++ D:\expdata\c3\SysEdit\31/l/after.java
@@ -34,1 +34,1 @@
-    fKeyBindingService = ps != null ? ps.getKeyBindingService() : null;
+    fHandlerService = ps != null ? (IHandlerService) ps.getService(IHandlerService.class) : null;
@@ -58,1 +58,1 @@
-            fKeyBindingService,
+            fHandlerService,
@@ -60,1 +60,2 @@
-            "org.eclipse.compare.copyAllLeftToRight"); //$NON-NLS-1$
+            "org.eclipse.compare.copyAllLeftToRight",
+            fActivations); //$NON-NLS-1$
@@ -74,1 +75,1 @@
-            fKeyBindingService,
+            fHandlerService,
@@ -76,1 +77,2 @@
-            "org.eclipse.compare.copyAllRightToLeft"); //$NON-NLS-1$
+            "org.eclipse.compare.copyAllRightToLeft",
+            fActivations); //$NON-NLS-1$
