--- /Users/yumeng/PycharmProjects/SysEdit/23/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/23/l/after.java
@@ -1,1 +1,1 @@
-public RecoveredElement add(FieldDeclaration fieldDeclaration, int bracketBalance) {
+public RecoveredElement add(FieldDeclaration fieldDeclaration, int bracketBalanceValue) {
@@ -13,1 +13,1 @@
-            return this.parent.add(fieldDeclaration, bracketBalance);
+            return this.parent.add(fieldDeclaration, bracketBalanceValue);
@@ -25,1 +25,1 @@
-            return this.parent.add(fieldDeclaration, bracketBalance);
+            return this.parent.add(fieldDeclaration, bracketBalanceValue);
