--- /Users/yumeng/PycharmProjects/SysEdit/22/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/22/l/after.java
@@ -22,1 +22,1 @@
-            codeStream.updateLastRecordedEndPC(currentScope, codeStream.position);
+            codeStream.recordPositionsFrom(codeStream.position, this.sourceEnd);
@@ -42,1 +42,1 @@
-            codeStream.updateLastRecordedEndPC(currentScope, codeStream.position);
+            codeStream.recordPositionsFrom(codeStream.position, this.sourceEnd);
@@ -70,1 +70,1 @@
-                codeStream.updateLastRecordedEndPC(currentScope, codeStream.position);
+                codeStream.recordPositionsFrom(codeStream.position, this.sourceEnd);
@@ -92,1 +92,1 @@
-            codeStream.updateLastRecordedEndPC(currentScope, codeStream.position);
+            codeStream.recordPositionsFrom(codeStream.position, this.sourceEnd);
