--- /Users/yumeng/PycharmProjects/SysEdit/15/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/15/l/after.java
@@ -1,2 +1,2 @@
-private org.eclipse.compare.contentmergeviewer.RGB getStrokeColor(Diff diff) {
-    boolean selected = (fCurrentDiff != null) && (fCurrentDiff.fParent == diff);
+private org.eclipse.compare.contentmergeviewer.RGB getStrokeColor(org.eclipse.compare.internal.merge.DocumentMerger.Diff diff) {
+    boolean selected = (fCurrentDiff != null) && (fCurrentDiff.getParent() == diff);
@@ -4,2 +4,2 @@
-        switch (diff.fDirection) {
-            case RangeDifference.RIGHT :
+        switch (diff.getKind()) {
+            case org.eclipse.compare.rangedifferencer.RangeDifference.RIGHT :
@@ -10,1 +10,1 @@
-            case RangeDifference.ANCESTOR :
+            case org.eclipse.compare.rangedifferencer.RangeDifference.ANCESTOR :
@@ -12,1 +12,1 @@
-            case RangeDifference.LEFT :
+            case org.eclipse.compare.rangedifferencer.RangeDifference.LEFT :
@@ -17,1 +17,1 @@
-            case RangeDifference.CONFLICT :
+            case org.eclipse.compare.rangedifferencer.RangeDifference.CONFLICT :
