--- /Users/yumeng/PycharmProjects/SysEdit/32/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/32/l/after.java
@@ -1,2 +1,2 @@
-private void paintSides(org.eclipse.swt.graphics.GC g, org.eclipse.compare.internal.MergeSourceViewer tp, org.eclipse.swt.widgets.Canvas canvas, boolean right) {
-    org.eclipse.swt.widgets.Display display = canvas.getDisplay();
+private void paintSides(GC g, MergeSourceViewer tp, Canvas canvas, boolean right) {
+    Display display = canvas.getDisplay();
@@ -5,1 +5,1 @@
-    org.eclipse.swt.graphics.Point size = canvas.getSize();
+    Point size = canvas.getSize();
@@ -25,1 +25,1 @@
-        org.eclipse.swt.graphics.Point region = new org.eclipse.swt.graphics.Point(0, 0);
+        Point region = new Point(0, 0);
@@ -50,1 +50,2 @@
-            g.setLineWidth(LW);
+            /* LW */
+            g.setLineWidth(0);
