--- /Users/yumeng/PycharmProjects/SysEdit/21/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/21/l/after.java
@@ -1,3 +1,3 @@
-public synchronized void contentInserted(int offset, int length) {
-    if (positions.size() == 0)
-        return;
+void contentInserted(int offset, int length) {
+    if (offset > this.offset)
+        throw new ArrayIndexOutOfBoundsException();
@@ -5,8 +5,2 @@
-    /* get all positions from offset to the end, inclusive */
-    Iterator iter = positions.tailMap(new PosBottomHalf(offset)).keySet().iterator();
-    iteration = true;
-    while (iter.hasNext()) {
-        PosBottomHalf bh = ((PosBottomHalf) (iter.next()));
-        bh.offset += length;
-    } 
-    iteration = false;
+    this.offset += length;
+    checkInvariants();
