--- /Users/yumeng/PycharmProjects/SysEdit/20/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/20/l/after.java
@@ -9,1 +9,1 @@
-        String actionName = ((String) (st.nextToken()));
+        String actionName = st.nextToken();
@@ -31,1 +31,0 @@
-    buttons.add(javax.swing.Box.createGlue());
@@ -33,1 +32,2 @@
-    add = new javax.swing.JButton(jEdit.getProperty("options.context.add"));
+    add = new RolloverButton(GUIUtilities.loadIcon("Plus.png"));
+    add.setToolTipText(jEdit.getProperty("common.add"));
@@ -37,1 +37,2 @@
-    remove = new javax.swing.JButton(jEdit.getProperty("options.context.remove"));
+    remove = new RolloverButton(GUIUtilities.loadIcon("Minus.png"));
+    remove.setToolTipText(jEdit.getProperty("common.remove"));
@@ -41,1 +42,2 @@
-    moveUp = new javax.swing.JButton(jEdit.getProperty("options.context.moveUp"));
+    moveUp = new RolloverButton(GUIUtilities.loadIcon("ArrowU.png"));
+    moveUp.setToolTipText(jEdit.getProperty("common.moveUp"));
@@ -45,1 +47,2 @@
-    moveDown = new javax.swing.JButton(jEdit.getProperty("options.context.moveDown"));
+    moveDown = new RolloverButton(GUIUtilities.loadIcon("ArrowD.png"));
+    moveDown.setToolTipText(jEdit.getProperty("common.moveDown"));
