--- /Users/yumeng/PycharmProjects/SysEdit/8/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/8/r/after.java
@@ -2,1 +2,2 @@
-    if (fContainer == null) {
+    org.eclipse.ui.services.IServiceLocator serviceLocator = fContainer.getServiceLocator();
+    if ((serviceLocator == null) && (!fContainerProvided)) {
@@ -6,1 +7,1 @@
-    return fContainer.getServiceLocator();
+    return serviceLocator;
