--- /Users/yumeng/PycharmProjects/SysEdit/26/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/26/r/after.java
@@ -6,0 +6,2 @@
+        org.eclipse.core.runtime.IConfigurationElement configurationElement = null;
+        String id = null;
@@ -7,3 +9,2 @@
-            org.eclipse.core.runtime.IConfigurationElement configurationElement = infos[i];
-            String id = configurationElement.getAttribute("id");// $NON-NLS-1$
-
+            configurationElement = infos[i];
+            id = configurationElement.getAttribute(ID);
