--- D:\expdata\c3\SysEdit\26/r/before.java
+++ D:\expdata\c3\SysEdit\26/r/after.java
@@ -10,0 +10,2 @@
+      IConfigurationElement configurationElement = null;
+      String id = null;
@@ -11,2 +13,2 @@
-        IConfigurationElement configurationElement = infos[i];
-        String id = configurationElement.getAttribute("id"); // $NON-NLS-1$
+        configurationElement = infos[i];
+        id = configurationElement.getAttribute(ID);
