--- D:\expdata\c3\SysEdit\39/r/before.java
+++ D:\expdata\c3\SysEdit\39/r/after.java
@@ -9,1 +9,1 @@
-    if (fChangeDiffs != null) {
+    if (fMerger.hasChanges()) {
@@ -13,3 +13,3 @@
-      Iterator e = fChangeDiffs.iterator();
-      while (e.hasNext()) {
-        Diff diff = (Diff) e.next();
+      char leg = getLeg(tp);
+      for (Iterator iterator = fMerger.changesIterator(); iterator.hasNext(); ) {
+        Diff diff = (Diff) iterator.next();
@@ -20,1 +20,1 @@
-        tp.getLineRange(diff.getPosition(tp), region);
+        tp.getLineRange(diff.getPosition(leg), region);
