--- /Users/yumeng/PycharmProjects/SysEdit/26/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/26/l/after.java
@@ -6,0 +6,2 @@
+        org.eclipse.core.runtime.IConfigurationElement configurationElement = null;
+        String attr = null;
@@ -7,2 +9,2 @@
-            org.eclipse.core.runtime.IConfigurationElement configurationElement = infos[i];
-            String attr = configurationElement.getAttribute("attribute");// $NON-NLS-1$
+            configurationElement = infos[i];
+            attr = configurationElement.getAttribute("attribute");// $NON-NLS-1$
