--- /Users/yumeng/PycharmProjects/SysEdit/43/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/43/r/after.java
@@ -2,2 +2,2 @@
-    if (element instanceof MyDiffNode) {
-        MyDiffNode myDiffNode = ((MyDiffNode) (element));
+    if (element instanceof org.eclipse.compare.internal.patch.PatcherDiffNode) {
+        org.eclipse.compare.internal.patch.PatcherDiffNode myDiffNode = ((org.eclipse.compare.internal.patch.PatcherDiffNode) (element));
@@ -7,1 +7,1 @@
-            switch (diff.getType()) {
+            switch (diff.getDiffType()) {
