--- /Users/yumeng/PycharmProjects/SysEdit/36/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/36/r/after.java
@@ -7,4 +7,11 @@
-    String leftLabel = PatchMessages.HunkMergePageInput_WorkspaceCopy;
-    cc.setLeftLabel(leftLabel);
-    String rightLabel = PatchMessages.HunkMergePageInput_OrphanedHunk;
-    cc.setRightLabel(rightLabel);
+    if (config != null) {
+        cc.setLeftLabel(config.getLeftLabel(config));
+        cc.setLeftImage(config.getLeftImage(config));
+        cc.setRightLabel(config.getRightLabel(config));
+        cc.setRightImage(config.getRightImage(config));
+    } else {
+        String leftLabel = PatchMessages.HunkMergePageInput_WorkspaceCopy;
+        cc.setLeftLabel(leftLabel);
+        String rightLabel = PatchMessages.HunkMergePageInput_OrphanedHunk;
+        cc.setRightLabel(rightLabel);
+    }
