--- /Users/yumeng/PycharmProjects/SysEdit/35/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/35/l/after.java
@@ -1,3 +1,3 @@
-protected void createToolItems(ToolBarManager tbm) {
-    IWorkbenchPartSite ps = Utilities.findSite(fComposite);
-    fHandlerService = (ps != null) ? ((org.eclipse.ui.handlers.IHandlerService) (ps.getService(org.eclipse.ui.handlers.IHandlerService.class))) : null;
+protected void createToolItems(org.eclipse.jface.action.ToolBarManager tbm) {
+    org.eclipse.ui.services.IServiceLocator locator = getCompareConfiguration().getContainer().getServiceLocator();
+    fHandlerService = (locator != null) ? ((org.eclipse.ui.handlers.IHandlerService) (locator.getService(org.eclipse.ui.handlers.IHandlerService.class))) : null;
@@ -6,1 +6,1 @@
-    Action ignoreAncestorAction = new Action() {
+    org.eclipse.jface.action.Action ignoreAncestorAction = new org.eclipse.jface.action.Action() {
@@ -8,2 +8,7 @@
-            setIgnoreAncestor(!fIgnoreAncestor);
-            Utilities.initToggleAction(this, getResourceBundle(), ignoreAncestorActionKey, fIgnoreAncestor);
+            // First make sure the ancestor is hidden
+            if (!isIgnoreAncestor())
+                getCompareConfiguration().setProperty(ICompareUIConstants.PROP_ANCESTOR_VISIBLE, Boolean.FALSE);
+
+            // Then set the property to ignore the ancestor
+            getCompareConfiguration().setProperty(ICompareUIConstants.PROP_IGNORE_ANCESTOR, Boolean.valueOf(!isIgnoreAncestor()));
+            org.eclipse.compare.internal.Utilities.initToggleAction(this, getResourceBundle(), ignoreAncestorActionKey, isIgnoreAncestor());
@@ -12,4 +17,4 @@
-    ignoreAncestorAction.setChecked(fIgnoreAncestor);
-    Utilities.initAction(ignoreAncestorAction, getResourceBundle(), ignoreAncestorActionKey);
-    Utilities.initToggleAction(ignoreAncestorAction, getResourceBundle(), ignoreAncestorActionKey, fIgnoreAncestor);
-    fIgnoreAncestorItem = new ActionContributionItem(ignoreAncestorAction);
+    ignoreAncestorAction.setChecked(isIgnoreAncestor());
+    org.eclipse.compare.internal.Utilities.initAction(ignoreAncestorAction, getResourceBundle(), ignoreAncestorActionKey);
+    org.eclipse.compare.internal.Utilities.initToggleAction(ignoreAncestorAction, getResourceBundle(), ignoreAncestorActionKey, isIgnoreAncestor());
+    fIgnoreAncestorItem = new org.eclipse.jface.action.ActionContributionItem(ignoreAncestorAction);
@@ -19,2 +24,2 @@
-    tbm.add(new Separator());
-    Action a = new Action() {
+    tbm.add(new org.eclipse.jface.action.Separator());
+    org.eclipse.jface.action.Action a = new org.eclipse.jface.action.Action() {
@@ -22,1 +27,1 @@
-            navigate(true, true, true);
+            navigate(true, true, isStepIntoEnabled());
@@ -25,1 +30,1 @@
-    Utilities.initAction(a, getResourceBundle(), "action.NextDiff.");// $NON-NLS-1$
+    org.eclipse.compare.internal.Utilities.initAction(a, getResourceBundle(), "action.NextDiff.");// $NON-NLS-1$
@@ -27,1 +32,1 @@
-    fNextItem = new ActionContributionItem(a);
+    fNextItem = new org.eclipse.jface.action.ActionContributionItem(a);
@@ -30,1 +35,1 @@
-    Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.selectNextChange", fActivations);// $NON-NLS-1$
+    org.eclipse.compare.internal.Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.selectNextChange", fActivations);// $NON-NLS-1$
@@ -32,1 +37,1 @@
-    a = new Action() {
+    a = new org.eclipse.jface.action.Action() {
@@ -34,1 +39,1 @@
-            navigate(false, true, true);
+            navigate(false, true, isStepIntoEnabled());
@@ -37,1 +42,1 @@
-    Utilities.initAction(a, getResourceBundle(), "action.PrevDiff.");// $NON-NLS-1$
+    org.eclipse.compare.internal.Utilities.initAction(a, getResourceBundle(), "action.PrevDiff.");// $NON-NLS-1$
@@ -39,1 +44,1 @@
-    fPreviousItem = new ActionContributionItem(a);
+    fPreviousItem = new org.eclipse.jface.action.ActionContributionItem(a);
@@ -42,1 +47,1 @@
-    Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.selectPreviousChange", fActivations);// $NON-NLS-1$
+    org.eclipse.compare.internal.Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.selectPreviousChange", fActivations);// $NON-NLS-1$
@@ -44,1 +49,7 @@
-    CompareConfiguration cc = getCompareConfiguration();
+    a = new org.eclipse.compare.internal.ChangePropertyAction(getResourceBundle(), getCompareConfiguration(), "action.StepInto.", STEP_INTO_PROPERTY);// $NON-NLS-1$
+
+    fStepIntoItem = new org.eclipse.jface.action.ActionContributionItem(a);
+    tbm.appendToGroup("navigation", fStepIntoItem);// $NON-NLS-1$
+
+    // Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.stepIntoChange", fActivations);	//$NON-NLS-1$
+    org.eclipse.compare.CompareConfiguration cc = getCompareConfiguration();
@@ -46,1 +57,1 @@
-        a = new Action() {
+        a = new org.eclipse.jface.action.Action() {
@@ -51,1 +62,1 @@
-        Utilities.initAction(a, getResourceBundle(), "action.CopyDiffLeftToRight.");// $NON-NLS-1$
+        org.eclipse.compare.internal.Utilities.initAction(a, getResourceBundle(), "action.CopyDiffLeftToRight.");// $NON-NLS-1$
@@ -53,1 +64,1 @@
-        fCopyDiffLeftToRightItem = new ActionContributionItem(a);
+        fCopyDiffLeftToRightItem = new org.eclipse.jface.action.ActionContributionItem(a);
@@ -57,1 +68,1 @@
-        Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.copyLeftToRight", fActivations);// $NON-NLS-1$
+        org.eclipse.compare.internal.Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.copyLeftToRight", fActivations);// $NON-NLS-1$
@@ -61,1 +72,1 @@
-        a = new Action() {
+        a = new org.eclipse.jface.action.Action() {
@@ -66,1 +77,1 @@
-        Utilities.initAction(a, getResourceBundle(), "action.CopyDiffRightToLeft.");// $NON-NLS-1$
+        org.eclipse.compare.internal.Utilities.initAction(a, getResourceBundle(), "action.CopyDiffRightToLeft.");// $NON-NLS-1$
@@ -68,1 +79,1 @@
-        fCopyDiffRightToLeftItem = new ActionContributionItem(a);
+        fCopyDiffRightToLeftItem = new org.eclipse.jface.action.ActionContributionItem(a);
@@ -72,1 +83,1 @@
-        Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.copyRightToLeft", fActivations);// $NON-NLS-1$
+        org.eclipse.compare.internal.Utilities.registerAction(fHandlerService, a, "org.eclipse.compare.copyRightToLeft", fActivations);// $NON-NLS-1$
