--- /Users/yumeng/PycharmProjects/SysEdit/32/r/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/32/r/after.java
@@ -1,2 +1,2 @@
-private void paintCenter(org.eclipse.swt.widgets.Canvas canvas, org.eclipse.swt.graphics.GC g) {
-    org.eclipse.swt.widgets.Display display = canvas.getDisplay();
+private void paintCenter(Canvas canvas, GC g) {
+    Display display = canvas.getDisplay();
@@ -9,1 +9,1 @@
-    org.eclipse.swt.graphics.Point size = canvas.getSize();
+    Point size = canvas.getSize();
@@ -27,1 +27,1 @@
-        org.eclipse.swt.graphics.Point region = new org.eclipse.swt.graphics.Point(0, 0);
+        Point region = new Point(0, 0);
@@ -57,2 +57,2 @@
-            org.eclipse.swt.graphics.Color fillColor = getColor(display, getFillColor(diff));
-            org.eclipse.swt.graphics.Color strokeColor = getColor(display, getStrokeColor(diff));
+            Color fillColor = getColor(display, getFillColor(diff));
+            Color strokeColor = getColor(display, getStrokeColor(diff));
@@ -66,1 +66,2 @@
-                g.setLineWidth(LW);
+                /* LW */
+                g.setLineWidth(0);
@@ -83,1 +84,2 @@
-                g.setLineWidth(LW);
+                /* LW */
+                g.setLineWidth(0);
@@ -99,1 +101,2 @@
-                g.setLineWidth(LW);
+                /* LW */
+                g.setLineWidth(0);
