--- D:\expdata\c3\SysEdit\33/r/before.java
+++ D:\expdata\c3\SysEdit\33/r/after.java
@@ -12,14 +12,0 @@
-    if (IFile.class.equals(adapter)) {
-      IProgressMonitor pm = new NullProgressMonitor();
-      try {
-        flushViewer(fStructureInputPane, pm);
-        flushViewer(fStructurePane1, pm);
-        flushViewer(fStructurePane2, pm);
-        flushViewer(fContentInputPane, pm);
-      } catch (CoreException e) {
-        CompareUIPlugin.log(e);
-      }
-      IFile[] files = (IFile[]) getAdapter(IFile[].class);
-      if (files != null && files.length > 0) return files[0];
-      return null;
-    }
