--- /Users/yumeng/PycharmProjects/SysEdit/7/l/before.java
+++ /Users/yumeng/PycharmProjects/SysEdit/7/l/after.java
@@ -3,21 +3,1 @@
-        org.eclipse.core.runtime.jobs.Job job = new org.eclipse.core.runtime.jobs.Job("Opening Compare Editor") {
-            protected org.eclipse.compare.internal.IStatus run(IProgressMonitor monitor) {
-                IStatus status = prepareInput(input, monitor);
-                if (status.isOK()) {
-                    internalOpenEditor(input, page, editor);
-                    return Status.OK_STATUS;
-                }
-                if (status.getCode() == NO_DIFFERENCE) {
-                    org.eclipse.jface.dialogs.MessageDialog.openInformation(getShell(), Utilities.getString("CompareUIPlugin.dialogTitle"), Utilities.getString("CompareUIPlugin.noDifferences"));// $NON-NLS-1$//$NON-NLS-2$
-
-                    return Status.OK_STATUS;
-                }
-                return status;
-            }
-
-            public boolean belongsTo(Object family) {
-                return input.belongsTo(family);
-            }
-        };
-        job.setUser(true);
-        job.schedule();
+        openEditorInBackground(input, page, editor);
