--- D:\expdata\c3\SysEdit\7/l/before.java
+++ D:\expdata\c3\SysEdit\7/l/after.java
@@ -5,25 +5,1 @@
-      Job job =
-          new Job("Opening Compare Editor") {
-            protected IStatus run(IProgressMonitor monitor) {
-              IStatus status = prepareInput(input, monitor);
-              if (status.isOK()) {
-                internalOpenEditor(input, page, editor);
-                return Status.OK_STATUS;
-              }
-              if (status.getCode() == NO_DIFFERENCE) {
-                MessageDialog.openInformation(
-                    getShell(),
-                    Utilities.getString("CompareUIPlugin.dialogTitle"),
-                    Utilities.getString(
-                        "CompareUIPlugin.noDifferences")); //$NON-NLS-1$//$NON-NLS-2$
-                return Status.OK_STATUS;
-              }
-              return status;
-            }
-
-            public boolean belongsTo(Object family) {
-              return input.belongsTo(family);
-            }
-          };
-      job.setUser(true);
-      job.schedule();
+      openEditorInBackground(input, page, editor);
