--- D:\expdata\c3\SysEdit\24/l/before.java
+++ D:\expdata\c3\SysEdit\24/l/after.java
@@ -3,1 +3,1 @@
-    String listKey = element.getAttribute("key");
+    String listKey = element.getAttribute(KEY);
@@ -7,0 +7,2 @@
+    Node node = null;
+    Element selement = null;
@@ -8,6 +10,4 @@
-      Node node = nodeList.item(i);
-      short type = node.getNodeType();
-      if (type == Node.ELEMENT_NODE) {
-        Element subElement = (Element) node;
-        String nodeName = subElement.getNodeName();
-        if (!nodeName.equalsIgnoreCase("listEntry")) {
+      node = nodeList.item(i);
+      if (node.getNodeType() == Node.ELEMENT_NODE) {
+        selement = (Element) node;
+        if (!selement.getNodeName().equalsIgnoreCase(LIST_ENTRY)) {
@@ -16,2 +16,1 @@
-        String value = getValueAttribute(subElement);
-        list.add(value);
+        list.add(getValueAttribute(selement));
